internal/tools/go.mod

```diff
@@ -16,6 +16,7 @@ require (
 	go.opentelemetry.io/build-tools/issuegenerator v0.12.0
 	go.opentelemetry.io/build-tools/multimod v0.13.0
 	go.opentelemetry.io/collector/cmd/builder v0.96.0
+	go.opentelemetry.io/collector/cmd/mdatagen v0.96.1-0.20240304230612-89dcb871686b
```

Comment:
Do we want this to be the same version used by the builder?

Reply:
Yeah, I'm actually doing the core upgrade in a separate PR

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31612